docs/src/modules/components/Demo.js

```diff
@@ -282,7 +357,363 @@ function Demo(props) {
     handleMoreClose();
   };
 
+  const createHandleCodeSourceLink = (anchor) => async () => {
+    try {
+      await copy(`${window.location.href.split('#')[0]}#${anchor}`);
+      setSnackbarMessage(t('copiedSourceLink'));
+      setSnackbarOpen(true);
+    } finally {
+      handleMoreClose();
+    }
+  };
+
+  const [sourceHintSeen, setSourceHintSeen] = React.useState(false);
+  React.useEffect(() => {
+    setSourceHintSeen(getCookie('sourceHintSeen'));
+  }, []);
+  const handleCodeOpenClick = () => {
+    document.cookie = `sourceHintSeen=true;path=/;max-age=31536000`;
+    onCodeOpenChange();
+    setSourceHintSeen(true);
+  };
+
+  function handleResetFocusClick() {
+    initialFocusRef.current.focusVisible();
+  }
+
   const showSourceHint = demoHovered && !sourceHintSeen;
+
+  let showCodeLabel;
+  if (codeOpen) {
+    showCodeLabel = showPreview ? t('hideFullSource') : t('hideSource');
+  } else {
+    showCodeLabel = showPreview ? t('showFullSource') : t('showSource');
+  }
+
+  const atLeastMediumViewport = useMediaQuery((theme) => theme.breakpoints.up('sm'));
```

Comment:
Considering https://material-ui.com/customization/breakpoints/

> xs, extra-small: 0px
sm, small: 600px
md, medium: 960px
lg, large: 1280px
xl, extra-large: 1920px

```suggestion
  const atLeastSmallViewport = useMediaQuery((theme) => theme.breakpoints.up('sm'));
```

Reply:
I thought `up` is `>` not `>=`.